src/app/api/product/route.ts

```diff
@@ -0,0 +1,20 @@
+import { prisma } from '@/db' //client to talk to db
+import { Body, Response } from '@/shared/routes'
+import { NextResponse } from 'next/server'
+
+export async function POST(request: Request): Promise<Response<'/api/product'>> {
+  const body: Body<'/api/product'> = await request.json()
+  console.log(body)
+  const product = await prisma.product.create({
+    data: {
+      name: body.name,
+      displayName: body.displayName,
+      available: body.available,
+      price: body.price,
+      type: body.type,
+      families: body.families,
+      description: body.description,
+    },
```

Comment:
you can just pass the whole body rather than passing each attribute like so,

`data: body`